### PR TITLE
libjwt: init at 1.12.1

### PR DESCRIPTION
**Pull request by pnotequalnp**
###### Motivation for this change
Packages `libjwt` since the Haskell package `libjwt-typed` is currently broken because it depends on it.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libjwt/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, lib, fetchurl, autoreconfHook, pkg-config, jansson, openssl }:
```
Comment (humancalico) [inline pkgs/development/libraries/libjwt/default.nix:1]:
```suggestion
{ stdenv, lib, fetchFromGitHub, autoreconfHook, pkg-config, jansson, openssl }:
```

File: pkgs/development/libraries/libjwt/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+stdenv.mkDerivation rec {
+  name = "libjwt-${version}";
```
Comment (humancalico) [inline pkgs/development/libraries/libjwt/default.nix:4]:
```suggestion
  pname = "libjwt-${version}";
```

File: pkgs/development/libraries/libjwt/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  version = "v1.12.1";
```
Comment (humancalico) [inline pkgs/development/libraries/libjwt/default.nix:5]:
```suggestion
  version = "1.12.1";
```

File: pkgs/development/libraries/libjwt/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/benmcollins/libjwt/archive/${version}.tar.gz";
```
Comment (humancalico) [inline pkgs/development/libraries/libjwt/default.nix:8]:
```suggestion
  src = fetchFromGitHub {
    owner = "benmcollins";
    repo = "libjwt";
    rev = "v${version}";
```

File: pkgs/development/libraries/libjwt/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  };
+
+  buildInputs = [ jansson openssl ];
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+
+  meta = with lib; {
+    homepage = "https://github.com/benmcollins/libjwt";
+    description = "JWT C Library";
+    license = licenses.mpl20;
+    platforms = platforms.all;
```
Comment (humancalico) [inline pkgs/development/libraries/libjwt/default.nix:19]: Please add yourself as a maintainer the file https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix as a separate commit with the commit message `maintainers: add pnotequalnp` and add yourself as a maintainer of this library.

**Review by akshatagarwl — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libjwt/default.nix (hunk 1 above)
Comment (humancalico) [reply pkgs/development/libraries/libjwt/default.nix:1]: `fetchFromGitHub` is a better way to fetch than `fetchurl`
Source: https://nixos.org/manual/nixpkgs/stable/#sec-sources

File: pkgs/development/libraries/libjwt/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libjwt/default.nix:4]:
```suggestion
  pname = "libjwt";
```

**Comment by humancalico** [comment]
@pnotequalnp Can you squash this into just two commits?
1. `maintainers: add pnotequalnp`
2. `libjwt: init at 1.12.1`

**Pull request merged**